docarray/array/mixins/find.py

```diff
@@ -227,7 +233,7 @@ def find(
 
     @abc.abstractmethod
     def _find(
-        self, query: 'ArrayType', limit: int, **kwargs
+        self, query: 'ArrayType', limit: int, filter: Optional[Dict] = None, **kwargs
```

Comment:
`query` could be `Dict`

Reply:
no can't be, it is only used in https://github.com/jina-ai/docarray/blob/d0e18e860fcc693e27275e856b939e11da25d5cd/docarray/array/mixins/find.py#L180
and in case query or filter is dict, another method will be called

Reply:
should be all good now